frontend/src/features/nextcloud_integration/api/getImages.ts

```diff
@@ -26,6 +28,53 @@ export const getImageList = async (path: string): Promise<Array<string>> => {
   }
 };
 
+/**
+ * get list of available images at the public share path from Nextcloud
+ */
+export const getPublicFileList = async (publicShareToken: string): Promise<Array<string>> => {
+  const username = publicShareToken;
+  const password = publicShareToken;
+  try {
+    const response = await axios({
+      method: 'PROPFIND',
+      url: 'https://cloud.permaplant.net/public.php/webdav',
+      headers: {
+        Authorization: 'Basic ' + btoa(username + ':' + password),
+      },
+    });
+    const parser = new DOMParser();
+    const doc = parser.parseFromString(response.data, 'application/xml');
+    const imageUrls: Array<string> = [];
+    doc.documentElement.childNodes.forEach((value) => {
+      imageUrls.push(value.childNodes[0].childNodes[0].nodeValue ?? '');
+    });
+    return imageUrls;
```

Comment:
You could also use this XPath to find just the image urls
```ts
const parser = new DOMParser();
const doc = parser.parseFromString(response.data, "application/xml");
const nsResolver = doc.createNSResolver(doc.documentElement);
const imageUrls = [];

const result = doc.evaluate('//d:response[d:propstat/d:prop/d:getcontenttype[contains(text(), "image")]]/d:href/text()', doc, nsResolver, XPathResult.ANY_TYPE, null);
```

then, to get the actual files
```ts
const files = [];
let node = null;
while ((node = result.iterateNext())) {
  files.push(node.data);
}
```